realm/realm-library/src/objectServer/java/io/realm/internal/network/OkHttpAuthenticationServer.java

```diff
@@ -79,47 +81,63 @@ public AuthenticateResponse refreshUser(Token userToken, URI serverUrl, URL auth
     @Override
     public LogoutResponse logout(Token userToken, URL authenticationUrl) {
         try {
-            String requestBody = LogoutRequest.revoke(userToken).toJson();
-            return logout(buildLogoutUrl(authenticationUrl), requestBody);
+            String requestBody = LogoutRequest.create(userToken).toJson();
+            return logout(buildActionUrl(authenticationUrl, ACTION_LOGOUT), requestBody);
         } catch (Exception e) {
             return LogoutResponse.from(new ObjectServerError(ErrorCode.UNKNOWN, e));
         }
     }
 
-    private static URL buildLogoutUrl(URL authenticationUrl) {
+    @Override
+    public ChangePasswordResponse changePassword(Token userToken, String newPassword, URL authenticationUrl) {
+        try {
+            String requestBody = ChangePasswordRequest.create(userToken, newPassword).toJson();
+            return changePassword(buildActionUrl(authenticationUrl, ACTION_CHANGE_PASSWORD), requestBody);
+        } catch (Exception e) {
```

Comment:
I'll change to catch `Throwable` 👍 . Your right, it seems the Network thread pool is not propagating errors correctly. I'll create another issue for tracking and fixing that.